pkg/kubelet/lifecycle/predicate.go

```diff
@@ -277,12 +277,13 @@ func generalFilter(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) []Predica
 
 	// Check taint/toleration except for static pods
 	if !types.IsStaticPod(pod) {
-		_, isUntolerated := corev1.FindMatchingUntoleratedTaint(nodeInfo.Node().Spec.Taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
+		taint, isUntolerated := corev1.FindMatchingUntoleratedTaint(nodeInfo.Node().Spec.Taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
```

Comment:
Only the first untolerated one is returned here, should the others not be added to the reason?

Reply:
I only found the use of tainttoleration.ErrReasonNotMatch in test.go, but your concerns are valid, do you have any better suggestions?